buildSrc/subprojects/jvm/src/main/kotlin/gradlebuild/propagated-env-variables.kt

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gradlebuild
+
+import gradlebuild.basics.BuildEnvironment
+import org.gradle.api.tasks.testing.Test
+
+
+val propagatedEnvAllowList = listOf(
+    // Obviously necessary
+    "HOME",
+    "SHELL",
+    "TERM",
+
+    // Otherwise Windows complains "Unrecognized Windows Sockets error: 10106"
+    "SystemRoot",
+    "OS",
+
+    // For Android tests
+    "ANDROID_HOME",
+
+    // Used by Visual Studio
+    "USERNAME",
+    "USER",
+    "USERDOMAIN",
+    "USERPROFILE",
+    "LOCALAPPDATA",
+
+    // Used by Gradle test infrastructure
+    "REPO_MIRROR_URL",
+
+    // temp dir
+    "TMPDIR",
+    "TMP",
+    "TEMP",
+
+    // Seems important on Windows
+    "ALLUSERSPROFILE",
+    "PUBLIC",
+    "windir"
+)
+
+
+fun Test.configurePropagatedEnvVariables() {
+    if (BuildEnvironment.isCiServer) {
+        environment = System.getenv().entries.mapNotNull(::sanitize).toMap()
```

Comment:
Can we also have something that looks at the propagated environment variables and does a sanity check for "bad words"?

e.g., things like `BUILD_CACHE_PASSWORD` and `TEAMCITY_TOKEN` would make it through our filter, so I think we should make it so the build fails at this point if any env var contains (case insensitive) "password" or "token" so we're forced to pick a different name.

The failures at the test level below are OK, but it's too late at that point.

Reply:
A few candidates to blacklist:
- apikey
- secret
- auth
- credential